Disallow CUDA RPC to use new devices in output tensors

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54024 Disallow CUDA RPC to use new devices in output tensors**
* #54023 Disable CUDA RPC tests that use new device in user-function outputs

Differential Revision: [D27059108](https://our.internmc.facebook.com/intern/diff/D27059108)

mrshenli committed:
Disallow CUDA RPC to use new devices in output tensors

[ghstack-poisoned]

mrshenli committed:
Update on "Disallow CUDA RPC to use new devices in output tensors"




[ghstack-poisoned]

mrshenli committed:
Update on "Disallow CUDA RPC to use new devices in output tensors"


Differential Revision: [D27059108](https://our.internmc.facebook.com/intern/diff/D27059108)

[ghstack-poisoned]

mrshenli committed:
Update on "Disallow CUDA RPC to use new devices in output tensors"


Differential Revision: [D27059108](https://our.internmc.facebook.com/intern/diff/D27059108)

[ghstack-poisoned]

lw reviewed (approved):
Looks great, thanks so much!

mrshenli committed:
Update on "Disallow CUDA RPC to use new devices in output tensors"


Differential Revision: [D27059108](https://our.internmc.facebook.com/intern/diff/D27059108)

[ghstack-poisoned]

mrshenli committed:
Update on "Disallow CUDA RPC to use new devices in output tensors"


Differential Revision: [D27059108](https://our.internmc.facebook.com/intern/diff/D27059108)

[ghstack-poisoned]

mrshenli committed:
Update on "Disallow CUDA RPC to use new devices in output tensors"


Differential Revision: [D27059108](https://our.internmc.facebook.com/intern/diff/D27059108)

[ghstack-poisoned]

mrshenli committed:
Update on "Disallow CUDA RPC to use new devices in output tensors"


Differential Revision: [D27059108](https://our.internmc.facebook.com/intern/diff/D27059108)

[ghstack-poisoned]
